### PR TITLE
Accessibility options for indoor directions

### PR DESCRIPTION
**Pull request by gdzooks**
Things in this PR:
* Update to the UI to add a menu that allows you to open a popup with selectable accessibility options!
* The feature's logic!
* Tests!!

Things not in this PR:
* ~~Tests~~
* Refactoring that might have been template method but my brain refuses to read or write so here we are

Could not test the feature on my phone because master doesn't have indoor directions displaying on the correct floor for directions so... idk........

**NOTE**
If this doesn't build on your computer, I recommend you _clean_ the project and then build again. 
~~Also make sure your gradle is up to date :zzz:~~ Reverted the version dependency so you shouldn't have any trouble building because of a different gradle version.

### CODE REVIEW

File: app/src/main/java/com/example/campusguide/Accessibility.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        else allowStairs()
+    }
+
+    // private methods that change pathfinding
+    private fun forbidEscalators() {
+        forbiddenRooms.add("escalators")
+    }
+
+    private fun forbidElevators() {
+        forbiddenRooms.add("elevators")
```
Comment (sharon-ho) [inline app/src/main/java/com/example/campusguide/Accessibility.kt:33]: You can make strings `"elevators"`, `"stairs"` and `"elevators"` as string variables to pass.

File: app/src/main/java/com/example/campusguide/directions/indoor/SelectAccessibilityOptionsDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    private val directionsActivity: DirectionsActivity
+) : DialogFragment() {
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val inflater = requireActivity().layoutInflater
+        val view = inflater.inflate(R.layout.accessibility_select_in_directions, null)
+
+        // set title of dialog box
+        val builder: AlertDialog.Builder = AlertDialog.Builder(directionsActivity)
+            .setTitle("Select one or more options")
```
Comment (sharon-ho) [inline app/src/main/java/com/example/campusguide/directions/indoor/SelectAccessibilityOptionsDialogFragment.kt]: You can make the title string as a constant in `Constants`.

File: app/src/main/java/com/example/campusguide/directions/indoor/SelectAccessibilityOptionsDialogFragment.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+        builder.setView(view)
+        builder.create().window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+
+        builder.setPositiveButton("Confirm") { _, _ ->
+            handleOKSelected()
+        }.setNegativeButton("Cancel", null)
```
Comment (sharon-ho) [inline app/src/main/java/com/example/campusguide/directions/indoor/SelectAccessibilityOptionsDialogFragment.kt]: Use the string constants `CONFIRM_CHOICE` and `CANCEL_CHOICE` for the button setters.

File: app/src/main/res/layout/accessibility_select_in_directions.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    tools:visibility="visible">
+
+    <androidx.appcompat.widget.AppCompatCheckBox
+        android:id="@+id/escalators_checkbox"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Can't Use Escalators"/>
```
Comment (sharon-ho) [inline app/src/main/res/layout/accessibility_select_in_directions.xml]: Extract these hard-coded strings into `@string` resources xml (i.e. can't use escalators, elevators or stairs).

File: app/src/main/res/menu/accessibility_menu.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
+        android:id="@+id/access_popup"
+        android:title="Change Indoor Accessibility"/>
```
Comment (sharon-ho) [inline app/src/main/res/menu/accessibility_menu.xml]: Extract these hard-coded strings into `@string` resources xml (i.e. can't use escalators, elevators or stairs).

**Review by sharon-ho — CHANGES_REQUESTED** [review]
No further refactoring I can personally recommend. Just a couple changes to use constants for strings as suggested below.

**Comment by felixlapierre** [comment]
Currently investigating something that's preventing this from working properly

**Review by sharon-ho — APPROVED** [review]
Approving changes made!

**Pull request merged**